From 3027cb7141fb65cf3eeda69c688db8c4045e2d3f Mon Sep 17 00:00:00 2001
From: Giovanni Campagna <gcampagna-cNUdlRotFMnNLxjTenLetw@public.gmane.org>
Date: Sat, 12 Oct 2013 17:45:25 +0200
Subject: [PATCH] Allow using obexd without systemd in the user session

Not all sessions run systemd --user (actually, the majority
doesn't), so the dbus daemon must be able to spawn obexd
directly, and to do so it needs the full path of the daemon.
---
 Makefile.obexd                      | 4 ++--
 obexd/src/org.bluez.obex.service    | 4 ----
 obexd/src/org.bluez.obex.service.in | 4 ++++
 3 files changed, 6 insertions(+), 6 deletions(-)
 delete mode 100644 obexd/src/org.bluez.obex.service
 create mode 100644 obexd/src/org.bluez.obex.service.in

diff --git a/Makefile.obexd b/Makefile.obexd
index 3760867..142e7c3 100644
--- Makefile.obexd
+++ Makefile.obexd
@@ -2,12 +2,12 @@
 if SYSTEMD
 systemduserunitdir = @SYSTEMD_USERUNITDIR@
 systemduserunit_DATA = obexd/src/obex.service
+endif
 
 dbussessionbusdir = @DBUS_SESSIONBUSDIR@
 dbussessionbus_DATA = obexd/src/org.bluez.obex.service
-endif
 
-EXTRA_DIST += obexd/src/obex.service.in obexd/src/org.bluez.obex.service
+EXTRA_DIST += obexd/src/obex.service.in obexd/src/org.bluez.obex.service.in
 
 obex_plugindir = $(libdir)/obex/plugins
 
diff --git a/obexd/src/org.bluez.obex.service b/obexd/src/org.bluez.obex.service
deleted file mode 100644
index a538088..0000000
--- obexd/src/org.bluez.obex.service
+++ /dev/null
@@ -1,4 +0,0 @@
-[D-BUS Service]
-Name=org.bluez.obex
-Exec=/bin/false
-SystemdService=dbus-org.bluez.obex.service
diff --git a/obexd/src/org.bluez.obex.service.in b/obexd/src/org.bluez.obex.service.in
new file mode 100644
index 0000000..9c815f2
--- /dev/null
+++ obexd/src/org.bluez.obex.service.in
@@ -0,0 +1,4 @@
+[D-BUS Service]
+Name=org.bluez.obex
+Exec=@libexecdir@/obexd
+SystemdService=dbus-org.bluez.obex.service
-- 
1.8.3.1


